src/include/cursor.h

```diff
@@ -527,6 +527,36 @@ struct __wt_cursor_table {
     WT_CURSOR **idx_cursors;
 };
 
+struct __wt_cursor_version {
+    WT_CURSOR iface;
+
+    WT_CURSOR *hs_cursor;    /* Queries of history cursor. */
+    WT_CURSOR *table_cursor; /* Queries of regular table cursor. */
+    WT_UPDATE *next_upd;
+
+    /*
+     * While we are iterating through updates on the update list, we need to remember information
+     * about the previous update we have just traversed so that we can record this as part of the
+     * debug metadata in the version cursor's key.
+     */
+    volatile uint64_t upd_txnid;
+    /* The previous traversed udpate's durable_ts will become the durable_stop_ts. */
+    wt_timestamp_t upd_durable_stop_ts;
+    /* The previous traversed update's start_ts will become the stop_ts. */
+    wt_timestamp_t upd_stop_ts;
+
+#define WT_VERSION_UPDATE_CHAIN 0
```

Comment:
I created these macros to indicate where the version is. We should return them in the key as well.